### ISSUE TITLE
Support for getting the display name

### ISSUE THREAD

**Issue by fantasticmao** [electron/electron#25252]
### Preflight Checklist

* [x] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/master/CONTRIBUTING.md) for this project.
* [x] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
* [x] I have searched the issue tracker for a feature request that matches the one I want to file, without success.

### Problem Description

When using multiple displays, we need to provide the user with the name of the display to choose where to place the `BrowserWindow`, as described in this issue #6899. But so far(Electron ^8.4.1 version), only the id attribute in the Display object. Can it be supported in a future version, it will be very helpful.

https://www.electronjs.org/docs/api/structures/display

### Proposed Solution

I found that VS Code has implemented such features, you can see the following figure. However, I did not find the relevant code to set this application menu in the VS Code project: [VS Code implementation code](https://github.com/microsoft/vscode/blob/ae0dbb397dadbd2a504b7fd5990c7cc51077aa98/src/vs/platform/menubar/electron-main/menubar.ts#L530-L557). Maybe it is achieved through native code...?

![image](https://user-images.githubusercontent.com/20675747/91822536-8331f800-ec6a-11ea-8a8f-a27d32ed2269.png)

### Alternatives Considered

/

### Additional Information

/

**Comment by jediwade** [comment]
@fantasticmao Was your VS Code instance vanilla or did you have some extensions installed? I too would like this feature and have been going through the VS Code source to try and figure out what they did, but not having any luck. I can't find such an option on my Windows machine at home (two displays). Will look to see if it is available on my work Mac on Monday.

**Comment by fantasticmao** [comment]
@jediwade Sorry for reply late. Finally I found that this feature may be provided by macOS, because other applications on macOS also have this option.

<img width="982" alt="image" src="https://user-images.githubusercontent.com/20675747/130317753-ed936f15-085f-49df-9ec3-87c0956b264f.png">

**Comment by jediwade** [comment]
@fantasticmao No worries on the delay. I decided to start working on my own method of getting the name of whatever monitors are plugged into a computer. I have a Windows solution (requires PowerShell commands and NodeJS execFile method) and have only just started to look for a macOS solution.

**Comment by ragauskl** [comment]
You can get this information from EDID (Extended Display Identification Data), I believe that's the source of it. But not all screens like built-in or projectors might have the name, not exactly sure how different OS put a name together then, but it differs depending on OS.

Maybe there are better ways, but that's what I found/use for my project as electron did not provide this, in case it's of any help you can find EDIDs in:

* Windows: Registry 'HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Enum\DISPLAY\\\<type>\\\<swnummer>\Device Parameters' EDID value
* Linux: /sys/class/drm/card0-\<connector>/edid  (can also read from display directly but I didn't get that far yet/reading from file was good enough)
* Mac: in Objective C, using NSScreen 'NSScreenNumber' get service port, using that create NSDictionary, and that will have key 'IODisplayEDID' (or just get ProductName from created NSDictionary)

**Comment by ip4368** [comment]
@ragauskl How do you match the corresponding EDID with the Display object from Electron?

**Comment by ragauskl** [comment]
@ip4368 I don't as I moved from using electron call to writing native addon myself for list of reasons. But when I was looking for how to get display information I saw that electron just uses functions provided by chromium, I never finished/ended up finding how chromium creates id's for every display to match it to EDID I was getting, but I would guess that there must be some link between two that can be made if you dig deeper in chromium source code

However, not all displays will have IDs in which case you have order in which displays are returned or match by position/size, none of the options guarantee that match will be 100% exact I think. But if you write native addon yourself, then I don't see reason to use electron/match to electron results as it does not provide more than you can get yourself

**Comment by BitesPotatoBacks** [comment]
@ragauskl @fantasticmao on MacOS, if we want to implement a pure JS approach (that will work on all architectures), we can use `child_process` to execute the command `system_profiler -json SPDisplaysDataType` like so:
```js
child_process.exec("system_profiler -json SPDisplaysDataType", (error, stdout, stderr) => {...})
```
Here is an example output from my M1 Mac Mini.

```json
{
  "SPDisplaysDataType" : [
    {
      "_name" : "kHW_AppleM1Item",
      "spdisplays_metalfamily" : "spdisplays_mtlgpufamilyapple7",
      "spdisplays_ndrvs" : [
        {
          "_name" : "LG Ultra HD",
          "_spdisplays_display-product-id" : "5b08",
          "_spdisplays_display-serial-number" : "2ac6b",
          "_spdisplays_display-vendor-id" : "1e6d",
          "_spdisplays_display-week" : "10",
          "_spdisplays_display-year" : "2020",
          "_spdisplays_displayID" : "1",
          "_spdisplays_pixels" : "3840 x 2160",
          "_spdisplays_resolution" : "3840 x 2160 @ 60.00Hz",
          "spdisplays_ambient_brightness" : "spdisplays_yes",
          "spdisplays_main" : "spdisplays_yes",
          "spdisplays_mirror" : "spdisplays_off",
          "spdisplays_online" : "spdisplays_yes",
          "spdisplays_pixelresolution" : "spdisplays_2160p",
          "spdisplays_resolution" : "3840 x 2160 @ 60.00Hz",
          "spdisplays_rotation" : "spdisplays_supported"
        }
      ],
      "spdisplays_vendor" : "sppci_vendor_Apple",
      "sppci_bus" : "spdisplays_builtin",
      "sppci_cores" : "8",
      "sppci_device_type" : "spdisplays_gpu",
      "sppci_model" : "Apple M1"
    }
  ]
}                      
```
Just use JS to filter the JSON data for the parent `spdisplays_ndrvs` and grab the display name from the `_name` child contained in it.

Once you figure out a way to retrieve the current display's ID, you can compare the ID with all existing `_spdisplays_displayID` children, and if the ID matches, pull the display name from the child's `_name` sibling.

**Comment by dsanders11** [comment]
Closed by #36855.

**Comment by codemii** [comment]
Hi @fantasticmao , 
This solution to get display name using label is not working in windows machine. Do you have any suggestion for the same ?? 
@jediwade - did you find any solution for the same ? Thanks!
